src/go/configgenerator/filterconfig/update_proto_descriptor_test.go

```diff
@@ -0,0 +1,301 @@
+			service:   "",
+			inDesc:    "invalid proto descriptor",
+			wantError: "failed to unmarshal",
+		},
+		{
+			// ApiNames has different Full service name, protoDescriptor is not modified.
+			desc: "Wrong apiName, not override",
+			service: `
+http: {
+  rules: {
+    selector: "package.name.Service.Method"
+    post: "/v2/{name=*}"
+  }
+}`,
+			apiNames: []string{"package.name.WrongService"},
+			inDesc: `
+file: {
+  name: "proto_file_path"
+  package: "package.name"
+  service: {
+    name: "Service"
+    method: {
+      name: "Method"
+      options: {
+        [google.api.http]: {
+	  post: "/v1/{name=*}"
+	}
+      }
+    }
+  }
+}`,
+			wantDesc: `
+file: {
+  name: "proto_file_path"
+  package: "package.name"
+  service: {
+    name: "Service"
+    method: {
+      name: "Method"
+      options: {
+        [google.api.http]: {
+	  post: "/v1/{name=*}"
+	}
+      }
+    }
+  }
+}`,
+		},
+		{
+			// ProtoDescriptor doesnt have MethodOptions, the http rule is copied
+			desc: "Not method options",
+			service: `
+http: {
+  rules: {
+    selector: "package.name.Service.Method"
+    post: "/v2/{name=*}"
+  }
+}`,
+			apiNames: []string{"package.name.Service"},
+			inDesc: `
+file: {
+  name: "proto_file_path"
+  package: "package.name"
+  service: {
+    name: "Service"
+    method: {
+      name: "Method"
+    }
+  }
+}`,
+			wantDesc: `
+file: {
+  name: "proto_file_path"
+  package: "package.name"
+  service: {
+    name: "Service"
+    method: {
+      name: "Method"
+      options: {
+        [google.api.http]: {
+          selector: "package.name.Service.Method"
+	  post: "/v2/{name=*}"
+	}
+      }
+    }
+  }
+}`,
+		},
+		{
+			// ProtoDescriptor has an empty MethodOptions. the http rule is copied
+			desc: "Empty method options",
+			service: `
+http: {
+  rules: {
+    selector: "package.name.Service.Method"
+    post: "/v2/{name=*}"
+  }
+}`,
+			apiNames: []string{"package.name.Service"},
+			inDesc: `
+file: {
+  name: "proto_file_path"
+  package: "package.name"
+  service: {
+    name: "Service"
+    method: {
+      name: "Method"
+      options: {
+      }
+    }
+  }
+}`,
+			wantDesc: `
+file: {
+  name: "proto_file_path"
+  package: "package.name"
+  service: {
+    name: "Service"
+    method: {
+      name: "Method"
+      options: {
+        [google.api.http]: {
+          selector: "package.name.Service.Method"
+	  post: "/v2/{name=*}"
+	}
+      }
+    }
+  }
+}`,
+		},
+		{
+			// ProtoDescriptor has a different annotation, the http rule is coped
+			desc: "Basic overwritten case",
+			service: `
+http: {
+  rules: {
+    selector: "package.name.Service.Method"
+    post: "/v2/{name=*}"
+  }
+}`,
+			apiNames: []string{"package.name.Service"},
+			inDesc: `
+file: {
+  name: "proto_file_path"
+  package: "package.name"
+  service: {
+    name: "Service"
+    method: {
+      name: "Method"
+      options: {
+        [google.api.http]: {
+	  post: "/v1/abc/{name=*}"
```

Comment:
Instead of overwrite, should we place the copied http rule into `additional_bindings`?